File: src/main/java/me/nickp0is0n/easylocalize/utils/LocalizeExporter.java
package me.nickp0is0n.easylocalize.utils;

import me.nickp0is0n.easylocalize.models.LocalizedString;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.PrintWriter;
import java.util.List;

public class LocalizeExporter {
    public void toFile(List<LocalizedString> localizedStrings, File outputFile) throws FileNotFoundException {
        PrintWriter out = new PrintWriter(outputFile);
        localizedStrings.forEach((localizedString -> out.println(localizedString.toString() + "\n")));
        out.close();
    }
}

File: src/main/kotlin/me/nickp0is0n/easylocalize/utils/LocalizeExporter.kt
package me.nickp0is0n.easylocalize.utils

import me.nickp0is0n.easylocalize.models.LocalizedString
import java.io.File
import java.io.PrintWriter

class LocalizeExporter {
    fun toFile(localizedStrings: List<LocalizedString>, outputFile: File) {
        val writer = PrintWriter(outputFile)
        var lastMark: String? = null

        writer.use {
            localizedStrings.forEach {
                if (it.copyrightHeader != null) {
                    writer.println("/*${it.copyrightHeader}*/") // writing xcode copyright header on top
                }
                if (it.mark != lastMark && it.mark != null) {
                    lastMark = it.mark
                    writer.println("\n// MARK:${it.mark}\n")
                }
                writer.println(it)
            }
        }
    }
}